cpp/include/rmm/device_scalar.hpp

```diff
@@ -203,7 +252,15 @@ class device_scalar {
    */
   void set_value_async(value_type const& value, cuda_stream_view stream)
   {
-    _storage.set_element_async(0, value, stream);
+    if (_host_bounce_buffer) {
+      // Case: Copying with pinned host memory
+      **_host_bounce_buffer = value;
+      RMM_CUDA_TRY(cudaMemcpyAsync(
+        data(), *_host_bounce_buffer, sizeof(T), cudaMemcpyDefault, stream.value()));
+    } else {
+      // Case: Copying with pageable host memory — may trigger an implicit synchronization.
+      _storage.set_element_async(0, value, stream);
+    }
   }
 
   // Disallow passing literals to set_value to avoid race conditions where the memory holding the
```

Comment:
I believe you can do this too now? copy to bounce buffer immediately and then do the copy async

Reply:
I think @devavret is referring to the deleted function below.

Reply:
I get it now. Let’s first discuss where the buffer should be placed, as mentioned above. Once that’s settled, I can address this detailed point.